lib/Page.js

```diff
@@ -43,36 +43,26 @@ class Page extends EventEmitter {
    * @return {!Promise<!Page>}
    */
   static async create(client, target, ignoreHTTPSErrors, defaultViewport, screenshotTaskQueue) {
-
-    await client.send('Page.enable');
-    const {frameTree} = await client.send('Page.getFrameTree');
-    const page = new Page(client, target, frameTree, ignoreHTTPSErrors, screenshotTaskQueue);
+    const page = new Page(client, target, ignoreHTTPSErrors, screenshotTaskQueue);
     await Promise.all([
+      page._networkManager.initialize(),
```

Comment:
I have to think a lot about this. The frame manager can initialize before the network manager, and it has a network manager that it doesn’t know is uninitialized. I believe that it works, but it’s hard to reason about.

Reply:
> The frame manager can initialize before the network manager, and it has a network manager that it doesn’t know is uninitialized. I believe that it works, but it’s hard to reason about.

The following observations should help with reasoning:
- all state that depends on events is encapsulated inside the related manager.
- `page` instance is not exposed to a userland at this point - so all missed events is a natural race.

Also, there's no semantic change here: the initialization sequence maintains the same contracts as before. So if there's some confidence in the tip-of-tree code, it should be inheritable here.